relay-server/src/services/buffer/mod.rs

```diff
@@ -152,10 +145,12 @@ impl EnvelopeBufferService {
                         self.project_cache.send(UpdateProject(sampling_key));
                     }
                 }
-                self.changes = false;
+                // deprioritize the stack to prevent head-of-line blocking
+                self.buffer.mark_seen(&stack_key);
+                next_sleep = DEFAULT_SLEEP;
```

Comment:
Now that we cycle through unready stacks, we should not wait forever or until a new message comes along. Instead, we only sleep for a short default interval. As before, the sleep gets canceled when a new message comes in.